hotp/host/main.c

```diff
@@ -73,7 +73,7 @@ int main(void)
 	op.paramTypes = TEEC_PARAM_TYPES(TEEC_MEMREF_TEMP_INPUT,
 					 TEEC_NONE, TEEC_NONE, TEEC_NONE);
 	op.params[0].tmpref.buffer = K;
-	op.params[0].tmpref.size = sizeof(K);
+	op.params[0].tmpref.size = sizeof(K) - 1;
```

Comment:
`K` adds `'\0'` only for string output, and the `'\0'` does not belong to the content of the key.
https://github.com/linaro-swg/optee_examples/blob/a98d01e1b9168eaed96bcd0bac0df67c44a81081/hotp/host/main.c#L78

Reply:
Ok, well, it's not a string, it's a byte array, so there shouldn't be a NULL termination. So, in that case it's probably better to change the fprintf to a hexdump, or snprintf it or so. The reason this has worked so far, is because it's been plain printable ASCII numbers. Since a key in theory could be any (non-printable) value, a hexdump would probably make more sense than trying to use snprintf.